language/testing-infra/module-generation/src/generator.rs

```diff
@@ -335,3 +351,21 @@ impl<'a> ModuleGenerator<'a> {
         .gen()
     }
 }
+
+// Getter trait to be able to use [`StructTypeParameter`] and `(TypeVar, BTreeSet<Ability>)`
+// uniformily.
+trait TypeParamGetTypeVar {
```

Comment:
This is the same discussion as in the previous PR, luckily it is very local, and I thought it wouldn't be problematic to introduce the trait here because it's not public. I decided on the trait because the copy here is a bit more costly because `TypeVar`s are strings. The alternatives are:
* making `base_type` receive a `&[TypeVar]` which would require cloning all the type vars + the allocation of the array 
* making `base_type` receive a `&[&TypeVar]` which only require the extra array allocation

Reply:
I wouldn't be too worried about something like extra allocation, especially in this testing tool. I think it is better to stick to something less indirect, like some sort of `impl IntoIterator<Item = TypeVar>`